packages/drawnix/src/utils/image.ts

```diff
@@ -1,14 +1,79 @@
-import { getSelectedElements, PlaitBoard } from '@plait/core';
+import { getSelectedElements, PlaitBoard, ThemeColorMode } from '@plait/core';
 import { base64ToBlob, boardToImage, download } from './common';
 import { fileOpen } from '../data/filesystem';
 import { IMAGE_MIME_TYPES } from '../constants';
 import { insertImage } from '../data/image';
 
+// Get the actual background color from the board container
+const getThemeBackgroundColor = (board: PlaitBoard): string => {
+  try {
+    const boardContainer = PlaitBoard.getBoardContainer(board);
+
+    // Check multiple elements in the hierarchy for background color
+    const elementsToCheck = [
+      boardContainer,
+      boardContainer.parentElement,
+      boardContainer.closest('.drawnix'),
+      document.body,
+    ].filter(Boolean);
+
+    for (const element of elementsToCheck) {
+      if (element) {
+        const computedStyle = window.getComputedStyle(element as Element);
+        const backgroundColor = computedStyle.backgroundColor;
+
+        // If we find a non-transparent background color, use it
+        if (
+          backgroundColor &&
+          backgroundColor !== 'transparent' &&
+          backgroundColor !== 'rgba(0, 0, 0, 0)' &&
+          backgroundColor !== 'initial' &&
+          backgroundColor !== 'inherit'
+        ) {
+          return backgroundColor;
+        }
+      }
+    }
+
+    // If no background color found, use fallback
+    return getThemeBackgroundColorFallback(board.theme.themeColorMode);
+  } catch (error) {
+    console.warn(
+      'Could not get background color from DOM, using fallback',
+      error
+    );
+    return getThemeBackgroundColorFallback(board.theme.themeColorMode);
+  }
+};
+
+// Fallback theme background colors mapping
+const getThemeBackgroundColorFallback = (
```

Comment:
You can get the active themeColors data through `PlaitBoard.getThemeColors(board);`.

You can encapsulate e method exactly to get the background color of the drawing board in this way.